### ISSUE TITLE
EdgeView comparison only checks edge ID, not contents

### ISSUE THREAD

**Issue by leotrs** [xgi-org/xgi#194]
This is Bad.

```python3
In [11]: H1 = xgi.Hypergraph([{0, 1, 2}, {0}, {1, 2}])

In [12]: H2 = xgi.Hypergraph([{1, 2, 3}, {0}, {1, 3}])

In [13]: H1 == H2
Out[13]: False

In [14]: H1.edges == H2.edges
Out[14]: True
```

**Comment by leotrs** [comment]
OK to be honest, what I wrote in the title is my current hunch as to why this is happening.

**Comment by leotrs** [comment]
An easy (temporary?) fix is to encourage the user to do this instead:
```python3
H1.edges.members(dtype=dict) == H2.edges.members(dtype=dict)
```

**Comment by maximelucas** [comment]
Is this a bug though? `H.edges` contains edges IDs, so your example is not so surprising. 
Although a user might write
```python
In [14]: H1.edges == H2.edges
```
and expect
```python
Out[14]: False
```
assuming it's gonna compare edges members rather than IDs. 

We could implement `=` between `H.edges` as checking for the full dict {ID: members} as the fix you suggested. In both cases, we should clarify this in the doc.

**Comment by leotrs** [comment]
Yeah coming back to this the main problem I see is that the user may inadvertently compare two views from two different network instead of two views from the same network. And the question is whether we want EdgeViews to compare only based on the IDs rather than also take into account which network those IDs belong to.

**Comment by leotrs** [comment]
Okay I think my preference would be that comparing two views only works when they come from the same network. If the user wants to compare only the IDs contained, they can use `view.ids`. So we'd have:

```python
H1.edges == H2.edges
-> always False, regardless of edge IDs

H1.edges.ids == H2.edges.ids
-> may be True, if H1 and H2 have the same edge IDs

```

**Comment by maximelucas** [comment]
What do we want if the two have the same hyperedges (based on members) but different IDs?

**Comment by leotrs** [comment]
That sounds dangerously close to determining whether two hypergraphs are isomorphic :grin:

**Comment by leotrs** [comment]
I stand by the suggestion to make EdgeView comparison only work when the views belong to the same network. If not, it should always return False, and maybe even issue a warning?

If people agree, the implementation is really easy, just implement the `__eq__` method in EdgeView.

**Comment by nwlandry** [comment]
I agree. My thought is that for nodestats, this is equivalent to verifying that `H1.nodes(ids).memberships() == H2.nodes(ids).memberships()` I tried briefly messing around with this, but I got stuck with accessing members/memberships based on edge/nodestats.

**Comment by nwlandry** [comment]
@leotrs --- do you think that attributes should be included in the comparison?

**Comment by nwlandry** [comment]
Okay, I've thought more about this, and I think that including attributes in the comparison makes sense.

**Comment by nwlandry** [comment]
Okay, something that I'm running into is that if I implement this in the `IDView` base class, `NodeView` will also check whether the memberships are all the same instead of just checking the nodes. Thoughts on this?

**Comment by tlarock** [comment]
Based on our discussion in the meeting just now, I think there are two separate implementations here:

1. Which comparisons of hypergraphs/data structures do we want to implement? We definitely want users to have the option to make convenient and reliable comparisons between our data structures. Some options:

    - check for exact equality of `node_id=>node`, `edge_id=>edge`, and `attribute_name=>attribute_val` for all nodes/edges/etc.
    - relax the above by forgetting about the attributes, but still comparing mappings from `_id` to the actual node/edge data
    - further relax by forgetting about `edge_id`s and comparing the sets of edges themselves
        - similar to @leotrs's idea [here](https://github.com/xgi-org/xgi/issues/194#issuecomment-1271191732])
        - perhaps implemented as a set intersection if ordering of `members()` is not guaranteed (caveat that there will be multi-edge issues to work out, maybe using counters?)

Beyond these there is also of course isomorphism, but that should probably be left as a separate issue as I don't think anyone will expect `==` to mean isomorphism unless we tell them it does :)

2. Given the above possibilities (and any others), which do we want `H1 == H2` to implement? Same question for `H1.edges == H2.edges`, etc. Do we want to discourage or disallow any possibilities to avoid users making mistakes like comparing IDs when they want to know about labels?

Documentation will be important as some might also expect `==` to simply be object identity, i.e. are these the same exact object in memory? I wonder if there is some PEP standard out there to tell us what it is "supposed" to be...

@nwlandry let me know if this makes sense to you, happy to discuss the possibilities!
